### ISSUE TITLE
Support Grok

### ISSUE THREAD

**Issue by FloLecoeuche** [ollama/ollama#3222]
### What model would you like?

Please add [xai-org/grok-1](https://github.com/xai-org/grok-1) model to ollama.

**Comment by robmurrer** [comment]
is this possible? does it limit the hardware that it can run on? JAX is weird :)

**Comment by paboum** [comment]
> Exactly 8 GPUs are required due to the mesh shape assertion (1, 8). To run at native size, you will likely need 8x80GB GPUs (A100 80GB / H100 80GB).

It doesn't say that you need 80 GB VRAM to download it, just to run. Thus, no limitation for ollama, but perhaps a warning would be nice.

**Comment by 3Samourai** [comment]
It can be a very good addition to Ollama; just put a warning before downloading

**Comment by easp** [comment]
It’s a mediocre model with huge resource requirements. How is it a very good addition?

**Comment by not-nullptr** [comment]
agreed. i tried grok a while ago, it really gave no better results (and was a lot, lot more censored) than dolphin-mixtral (which ollama lists as 47B)

**Comment by 3Samourai** [comment]
> It’s a mediocre model with huge resource requirements. How is it a very good addition?

More choices are always better, and since it’s a V1, it’s natural if the performance is not the best

**Comment by dib258** [comment]
There's a smaller (quantized) version of Grok that got posted few hours ago. Could this be a better fit to run on smaller machine? and to be added to Ollama?

https://huggingface.co/Arki05/Grok-1-GGUF/tree/main

**Comment by joecryptotoo** [comment]
> There's a smaller (quantized) version of Grok that got posted few hours ago. Could this be a better fit to run on smaller machine? and to be added to Ollama?
> 
> https://huggingface.co/Arki05/Grok-1-GGUF/tree/main

I'm slowly pushing this to the repo here:

https://ollama.com/joefamous/grok-1

**Comment by joecryptotoo** [comment]
I was able to run the q6_K version with CPU on my 36 core Xeon. I got a little less than 1 token per second.

**Comment by joecryptotoo** [comment]
This might do it. https://github.com/ollama/ollama/pull/3348

**Comment by paboum** [comment]
ollama version is 0.1.29
```
time=2024-03-26T04:03:13.270+01:00 level=WARN source=llm.go:170 msg="Failed to load dynamic library /tmp/ollama1052153078/runners/cpu_avx2/libext_server.so  exception error loading model architecture: unknown model architecture: 'grok'"
```

**Comment by AWKohler** [comment]
> mic library /tmp/ollama1052153078/runners/cpu_avx2/libext_server.so  exception error loading model architectu

getting same issue

**Comment by Zhongyi-Lu** [comment]
> > There's a smaller (quantized) version of Grok that got posted few hours ago. Could this be a better fit to run on smaller machine? and to be added to Ollama?

> > https://huggingface.co/Arki05/Grok-1-GGUF/tree/main
> 
> I'm slowly pushing this to the repo here:
> 
> https://ollama.com/joefamous/grok-1

@joecryptotoo  Hi. Thanks for uploading models! I got an error saying

```
Error: exception error loading model architecture: unknown model architecture: 'grok'
```

Is it expected?

**Comment by taozhiyuai** [comment]
try to import gguf, but fail

<img width="823" alt="截屏2024-03-29 07 44 46" src="https://github.com/ollama/ollama/assets/146583103/0f037fac-f70c-47e0-9936-fffbe722b1fd">
<img width="1439" alt="截屏2024-03-29 07 45 24" src="https://github.com/ollama/ollama/assets/146583103/b61350a2-dab9-4dd5-9232-fe46f4efd101">

**Comment by taozhiyuai** [comment]
@joecryptotoo can you upload this Q1 version? https://hf-mirror.com/mradermacher/grok-1-i1-GGUF/tree/main

@Zhongyi-Lu

**Comment by userforsource** [comment]
https://github.com/xai-org/grok-1?tab=readme-ov-file
When will we get grok support in ollama.

**Comment by userforsource** [comment]
What is the support status of grok?

**Comment by userforsource** [comment]
What is the support status.

**Comment by olumolu** [comment]
When will we get gork support in ollama

**Comment by TraceRecursion** [comment]
When will we get gork support in ollama?

**Comment by Maltz42** [comment]
Adding my request for this as well.  Seems weird that one of the biggest AI players is missing support for a year.  It requires a lot of resources, but Llama 3.1 405B is there, and there are plenty of people with 256GB who could probably run a q4 in RAM.

**Comment by mat926** [comment]
Can we have grok3 support?

**Comment by olumolu** [comment]
> Can we have grok3 support?

Not a open weight model

**Comment by FearL0rd** [comment]
Grok2.1 is out 😉
